### PR TITLE
Fix tests collisions caused by multiple runs

### PR DESCRIPTION
**Pull request by Or-Geva**
* Add build delete service
* Delete published tests build
* Tests' build number changed from const-number to timestamp

- [x] All [tests](https://ci.appveyor.com/project/jfrog-ecosystem/build-info) passed. If this feature is not already covered by the tests, I added new tests.
-----
Related: https://github.com/jfrog/jenkins-artifactory-plugin/pull/529

### CODE REVIEW

File: build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/client/artifactory/services/DeleteBuilds.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+import java.io.IOException;
+
+/**
+ * Delete multiple build numbers of a certain build.
+ */
+public class DeleteBuilds extends VoidJFrogService {
+    public static final String DELETE_BUILDS_ENDPOINT = "api/build/";
+    private final String buildName;
+    private final boolean deleteArtifact;
```
Comment (eyalbe4) [inline build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/client/artifactory/services/DeleteBuilds.java:17]: deleteArtifact --> deleteArtifacts
(Let's also change the local variables' names).

File: build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/client/artifactory/ArtifactoryManager.java [hunk 2]
```diff
@@ -145,6 +145,16 @@ public void sendModuleInfo(Build build) throws IOException {
         sendModuleInfoService.execute(jfrogHttpClient);
     }
 
+    public void deleteBuilds(String buildName, String project, boolean deleteArtifact) throws IOException {
```
Comment (eyalbe4) [inline build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/client/artifactory/ArtifactoryManager.java:148]: deleteArtifact --> deleteArtifacts

File: build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/client/artifactory/ArtifactoryManager.java [hunk 3]
```diff
@@ -145,6 +145,16 @@ public void sendModuleInfo(Build build) throws IOException {
...
+        DeleteBuilds deleteBuildsService = new DeleteBuilds(buildName, project, deleteArtifact, log);
+        deleteBuildsService.execute(jfrogHttpClient);
+    }
+
+    public void deleteBuild(String buildName, String buildNumber, String project, boolean deleteArtifact) throws IOException {
```
Comment (eyalbe4) [inline build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/client/artifactory/ArtifactoryManager.java:153]: deleteArtifact --> deleteArtifacts

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: build-info-extractor-gradle/src/test/java/org/jfrog/gradle/plugin/artifactory/GradlePluginTest.java [hunk 4]
```diff
@@ -59,6 +60,9 @@ public void configurationsTest(String gradleVersion) throws IOException {
         BuildResult buildResult = runGradle(gradleVersion, envVars, false);
         // Check results
         checkBuildResults(artifactoryManager, buildResult, false, localRepo1);
+        // Cleanup
+        Build buildInfo = getBuildInfo(artifactoryManager, buildResult);
```
Comment (yahavi) [inline build-info-extractor-gradle/src/test/java/org/jfrog/gradle/plugin/artifactory/GradlePluginTest.java:64]: This REST is redundant. You already have the build name and the build number in the `buildResult`.
Also - shouldn't we make the build number random?

Comment (Or-Geva) [reply build-info-extractor-gradle/src/test/java/org/jfrog/gradle/plugin/artifactory/GradlePluginTest.java:64]: By default, BuildInfo is published with a build number that is the start date of the build.

File: build-info-extractor-gradle/src/test/java/org/jfrog/gradle/plugin/artifactory/Utils.java [hunk 5]
```diff
@@ -189,15 +190,24 @@ private static void assertRequestedBy(Build buildInfo) {
...
+    public static Build getBuildInfo(ArtifactoryManager artifactoryManager, BuildResult buildResult) throws IOException {
+        Pair<String,String> buildDetails = getBuildDetails(buildResult);
+        return artifactoryManager.getBuildInfo(buildDetails.getLeft(), buildDetails.getRight(), null);
+    }
+
+    /**
+     * @param buildResult Details of the build
+     * @return A pair of: Left item - buildResult's build name, Right item - buildResult's build number
+     */
+    public static Pair<String,String> getBuildDetails( BuildResult buildResult)  {
```
Comment (yahavi) [inline build-info-extractor-gradle/src/test/java/org/jfrog/gradle/plugin/artifactory/Utils.java:202]: Please run reformat on this file
```suggestion
    public static Pair<String, String> getBuildDetails(BuildResult buildResult)  {
```

**Review by yahavi — CHANGES_REQUESTED** [review]

**Pull request merged**